rusk/src/lib/http/event.rs

```diff
@@ -613,42 +613,45 @@ impl SessionId {
     }
 }
 
-/// A subscription to a contract event.
+/// A subscription to an event.
 ///
-/// The `topic` field is optional and can be used to filter events by topic. If
-/// the `topic` field is `None`, all events from the contract are sent.
+/// Subscriptions have data related to the component the subscriber wishes to
+/// subscribe to, the component targeted by the event (`contracts`,
+/// `transactions`, etc...) and an optional entity within the component that
+/// the event targets.
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
-pub struct ContractSubscription {
-    pub target: WrappedContractId,
+pub struct RuesSubscription {
     pub topic: String,
+    pub component: String,
+    pub entity: Option<String>,
 }
 
-impl ContractSubscription {
-    /// Parses a subscription from an url path split.
-    ///
-    /// Returns `None` if the path is invalid.
+impl RuesSubscription {
     pub fn parse_from_path_split(mut path_split: Split<char>) -> Option<Self> {
-        // Parse out the contract ID
-        let target_hex = path_split.next()?;
-        let target_bytes = hex::decode(target_hex).ok()?;
-        let mut target_array = [0u8; 32];
+        let topic = path_split.next()?.to_string();
+        let component = path_split.next()?.to_string();
 
-        if target_bytes.len() != 32 {
-            return None;
+        let mut entity = None;
+        if let Some(e) = path_split.next() {
+            entity = Some(e.to_string());
+            entity = Some(e.to_string());
```

Comment:
I think this leftover is the cause to not have clippy suggesting

```rust
let entity = path_split.next().map(|e| e.to_string());
```

Reply:
I'm making the change and clippy doesn't seem to complain. That said, it is a nice change, so I will do it.